packages/core/components/divider/index.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { color, space, themeGet } from "styled-system";
+import { Direction } from "../../common/direction";
+
+const grayLight1 = "#C3CADF";
+
+const HDivider = styled.hr`
+  ${space}
```

Comment:
I'm curious about the ordering of styled-system rules in general. Did you happen on any documentation/guidance on this?

What I ran into when putting the styled-system rules first was that any rule that covered some of the styled system rules ended up making those styled-system properties no longer have an effect. For example, declaring `${space};` and then on the next line having `margin-left: 5em;` would make the `mr` or `mx` property useless.

I think this problem is avoided if we do everything with default props but I also haven't done enough of this to know for sure. What's your feeling on this?

Reply:
I didn't notice any docs addressing this specifically, but it appears that order does matter and that, specifically, a declaration will override any previous declarations for the same property (including those from the styled-system function). This makes sense given it all just becomes CSS anyway, which behaves that way.

I think these weak rules of thumb make sense:
1. Put the styled-system functions first (for consistency), unless you have reason not to.
2. Use `defaultProps` for setting defaults, unless you have reason not to.

Eg, re (2), in this PR I couldn't use `defaultProps` for default `width` and `height` because those defaults differ depending on the `direction`. So I setup those defaults in the styled-component template literal instead, eg, `width: ${props => props.width || "100%"};`.

Reply:
:+1: This sounds good to me. Would you mind adding those rules-of-thumb to the README as well?